### ISSUE TITLE
Hight CPU usage when selecting

### ISSUE THREAD

**Issue by JAicewizard** [microsoft/vscode#78983]
Version: 1.38.0-insider
Commit: de320052255d136673fc5eee358f6e2551d6573e
Date: 2019-08-12T07:21:06.280Z
Electron: 4.2.9
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Linux x64 5.2.8-arch1-1-ARCH

Steps to Reproduce:
1. open any file
2. start selecting a bunch

This is a 100% clear install of insiders, no extensions, no settings changed.

~Attaching a .cpuprofile file is not supported by github, in what way should I upload the cpu profile?~
[vscodeinsiders.zip](https://github.com/microsoft/vscode/files/3496216/vscodeinsiders.zip)

**Comment by alexandrudima** [comment]
You can zip the .cpuprofile and attach it.

**Comment by JAicewizard** [comment]
[vscodeinsiders.zip](https://github.com/microsoft/vscode/files/3496216/vscodeinsiders.zip)
Thats the CPU profile.

**Comment by alexandrudima** [comment]
I looked at the CPU Profile and there is nothing out of the ordinary AFAICT. I see that a lot of frames have been rendered, presumably you have selected and moved your mouse around quite a lot. Rendering a frame appears to take a few ms in JavaScript (mostly under 10ms), which I think leaves enough time to make it to 60 FPS...

In other words, I think this is functioning as designed...

**Comment by JAicewizard** [comment]
I noticed that I had a lot of dropped frames when selecting rust code, and the CPU usage was at 10-20%
so I tried using insiders and its still 10-20% when selecting random text.
I think that that is quite high, it might hit the frame target, but thats about the same CPU usage as decoding 720p 60fps for me.

**Comment by alexandrudima** [comment]
I appreciate your point, but I think it is not helpful to compare apples and oranges.

We can talk about how inefficient it is for us to use a browser engine to render our editor, and how inefficient it is to create dom nodes with the text, which are then parsed from HTML, styled, layouted, painted and eventually transmitted via IPC to the GPU process, which finally composits the tiles in order for them to finally be visible... but IMHO such a discussion is not be really helpful.

In other words, the CPU consumption appears normal.

**Comment by JAicewizard** [comment]
I get that there is little that you can do about this, if you want to close this thats fine, but the issue itself is not resolved(I dont expect it to be, as there isnt a clear problem with the code).

I hope issues like this influence any future project wide decisions for vscode/electron.

**Comment by alexandrudima** [comment]
I will close this for now... We have ideas of reimplementing the editor using a `<canvas>` and textures for fonts. Those ideas might make things faster and save some CPU, but at this point those are ideas... It is also highly unlikely that we will rewrite our entire UI in something else than HTML/CSS, so migrating off Electron/Chromium is not really realistic IMHO...

**Comment by JAicewizard** [comment]
I wasnt saying to migrate off electron, just that electron might have possible changes that could improve rendering or something like that. I understand that it is highly unlikely that vscode will move off electron, and I certainly expect more would be needed to move off electron than some bad performance for some use-cases.

Moving over to '<canvas>' would be a project wide-decision I was talking about :)
